src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.CpuUtilizationReader.Windows.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    internal partial class ClrThreadPool
+    {
+        private class CpuUtilizationReader
+        {
+            private struct ProcessCpuInformation
+            {
+                public ulong idleTime;
+                public ulong kernelTime;
+                public ulong userTime;
+            }
+
+            private ProcessCpuInformation _cpuInfo = new ProcessCpuInformation();
+
+            public CpuUtilizationReader()
+            {
+                GetCpuUtilization(); // Call once to initialize the usage buffer
```

Comment:
I was just suggesting to do the first read in GateThreadStart, since that function is the only one that takes any readings. That's sort of what CoreCLR is doing, to ensure that every stored reading is over a 500 ms interval. Otherwise, the first stored reading would give us CPU utilization between process start time and first use of the thread pool.